packages/realm/src/index.ts

```diff
@@ -41,10 +45,13 @@ export type {
   CollectionChangeCallback,
   CollectionChangeSet,
   Configuration,
+  FlexibleSyncConfiguration,
   ObjectChangeCallback,
   ObjectChangeSet,
+  PartitionSyncConfiguration,
   RealmEventName,
   RealmListenerCallback,
+  SubscriptionOptions,
```

Comment:
Had to explicitly export `SubscriptionOptions` as a type, otherwise we get an error when running the tests in `packages/realm/`: *"The requested module './internal' does not provide an export named 'SubscriptionOptions'"*

Reply:
@kraenhansen Do you know why?

Reply:
@elle-j are you saying that it _had_ to be a _type_ export? Or that it simply had to be exported (type or not).

I believe we should be able to collapse these two lists of exports and not have to care if they're types or not.